deepmerge/strategy/core.py

```diff
@@ -1,20 +1,38 @@
+from __future__ import annotations
+
+import sys
+from typing import TYPE_CHECKING, List, Callable, Any, Union
+
+if sys.version_info >= (3, 10):
+    from typing import TypeAlias
+else:
+    from typing_extentions import TypeAlias
+
+import deepmerge.merger
 from ..exception import StrategyNotFound, InvalidMerge
-from ..compat import string_type
 
 STRATEGY_END = object()
 
+StrategyCallable: TypeAlias = "Callable[[deepmerge.merger.Merger, List, Any, Any], Any]"
+StrategyListInitable: TypeAlias = (
+    "Union[str, StrategyCallable, List[Union[str, StrategyCallable]]]"
+)
 
-class StrategyList(object):
 
-    NAME = None
+class StrategyList:
+    NAME: str
 
-    def __init__(self, strategy_list):
+    def __init__(self, strategy_list: StrategyListInitable) -> None:
         if not isinstance(strategy_list, list):
             strategy_list = [strategy_list]
-        self._strategies = [self._expand_strategy(s) for s in strategy_list]
+        self._strategies: List[StrategyCallable] = [
```

Comment:
nit: since you already use attribute parameters above (e.g. for `NAME`), I'd just annotate `_strategies` up there as well.

Reply:
The only reason I annotate `NAME` in that section is because it's a class attribute. For instance attributes I think it makes more sense to annotate them in `__init__`.

Reply:
I'm not strongly opinionated enough here to block the PR - but I think having the annotations in one place is nicer, although I do see the point about being unable to differentiate between class / object attributes.

Reply:
I can confirm that these are treated differently by type checkers. Defining a name in the class body specifies that is either a class or instance attribute (or class only using `typing.ClassVar`), whilst names defined in __init__ are instance only attributes. 

https://mypy.readthedocs.io/en/latest/class_basics.html#instance-and-class-attributes